kipoiseq/datasets/splicing.py

```diff
@@ -0,0 +1,258 @@
+        self.overhang = (0, 0)
+        self._isLast = False
+
+    @property
+    def isLast(self):
+        return self._isLast
+
+    @isLast.setter
+    def isLast(self, value):
+        self._isLast = value
+
+    @property
+    def isFirst(self):
+        return self.order == 1
+
+    @property
+    def grange(self):
+        return GenomicRanges(self.chrom,
+                             self.start,
+                             self.end,
+                             self.transcript_id,
+                             self.strand)
+
+    def __str__(self):
+        return '{0}_{1}_{2}:{3}'.format(
+            self.chrom, self.Exon_Start, self.Exon_End, self.strand)
+
+    @property
+    def to_dict(self):
+        return {'isLast': self.isLast,
+                'isFirst': self.isFirst,
+                'order': self.order,
+                'name': self.name,
+                'gene_id': self.gene_id,
+                'Exon_Start': self.Exon_Start,
+                'Exon_End': self.Exon_End,
+                'intronl_len': self.overhang[0],
+                'intronr_len': self.overhang[1],
+                'seqid': self.seqid,
+                'strand': self.strand,
+                'start': self.start,
+                'end': self.end}
+
+    @classmethod
+    def from_Feature(cls,
+                     feature,
+                     overhang=(0, 0)):
+        # convert gffutils.Feature to ExonInterval
+        iv = cls(seqid=feature.chrom,
+                 source=feature.source,
+                 start=feature.start,  # exon start
+                 end=feature.end,  # exon end
+                 strand=feature.strand,
+                 frame=feature.frame,
+                 attributes=feature.attributes,
+                 order=int(feature.attributes['exon_number'][0]))
+        if iv.strand == "+":
+            iv.start = iv.Exon_Start - overhang[0]
+            iv.end = iv.Exon_End + overhang[1]
+        else:
+            iv.start = iv.Exon_Start - overhang[1]
+            iv.end = iv.Exon_End + overhang[0]
+        iv.overhang = overhang
+        return iv
+
+    @classmethod
+    def from_exonfile(cls, exon, attributes, overhang=(0, 0)):
+        iv = cls(seqid=exon.CHROM,
+                 source='',
+                 start=exon.Exon_Start,  # exon start
+                 end=exon.Exon_End,  # exon end
+                 strand=exon.strand,
+                 frame='',
+                 attributes=attributes,
+                 order=attributes['order'])
+        if iv.strand == "+":
+            iv.start = iv.Exon_Start - overhang[0]
+            iv.end = iv.Exon_End + overhang[1]
+        else:
+            iv.start = iv.Exon_Start - overhang[1]
+            iv.end = iv.Exon_End + overhang[0]
+        iv.overhang = overhang
+        return iv
+
+    def get_seq(self, fasta, use_strand=True):
+        seq = self.sequence(fasta, use_strand=use_strand)
+        seq = seq.upper()
+        return seq
+
+
+def generateExons(gtf_file,
```

Comment:
Can you rename this into snake_case?

python naming convention:
- classes are CamelCase, class instances are camelCase
- functions, methods, attributes, variables etc are snake_case